iterable-extension/src/test/java/com/mparticle/ext/iterable/IterableExtensionTest.java

```diff
@@ -954,13 +950,80 @@ public void testHandleIterableResponseWith429() throws RetriableError {
+        Response itbl504 = Response.error(504, ResponseBody.create(
+                MediaType.parse("application/json; charset=utf-8"), "{}"));
+        IterableExtension.handleIterableResponse(itbl504, UUID.randomUUID());
+    }
+
     @Test(expected = RetriableError.class)
     public void testHandleIterableListResponseWith429() throws RetriableError {
         Response itbl492 = Response.error(429, ResponseBody.create(
                 MediaType.parse("application/json; charset=utf-8"), "{}"));
         IterableExtension.handleIterableListResponse(itbl492, UUID.randomUUID());
     }
 
+    @Test(expected = RetriableError.class)
+    public void testHandleIterableListResponseWith502() throws RetriableError {
+        Response itbl502 = Response.error(502, ResponseBody.create(
+                MediaType.parse("application/json; charset=utf-8"), "{}"));
+        IterableExtension.handleIterableListResponse(itbl502, UUID.randomUUID());
+    }
+
+    @Test(expected = RetriableError.class)
+    public void testHandleIterableListResponseWith504() throws RetriableError {
+        Response itbl504 = Response.error(504, ResponseBody.create(
+                MediaType.parse("application/json; charset=utf-8"), "{}"));
+        IterableExtension.handleIterableListResponse(itbl504, UUID.randomUUID());
+    }
+
+    @Test
+    public void testMakeIterableRequestWithSuccess() throws IOException {
+        Call call = createCallMockWithSuccessResponse();
+
+        makeIterableRequest(call, UUID.randomUUID());
+        Mockito.verify(call).execute();
+    }
+
+    @Test
+    public void testMakeIterableRequestWithListSuccess() throws IOException {
+        Call call = Mockito.mock(Call.class);
+        ListResponse listResponse = new ListResponse();
+        Response<ListResponse> response = Response.success(listResponse);
+        Mockito.when(call.execute()).thenReturn(response);
+
+        makeIterableRequest(call, UUID.randomUUID());
+        Mockito.verify(call).execute();
+    }
+
+    @Test(expected = RetriableError.class)
+    public void testMakeIterableRequestWithTimeout() throws IOException {
+        Call call = Mockito.mock(Call.class);
+        HttpUrl url = new HttpUrl.Builder()
+                .scheme("https")
+                .host("api.iterable.com")
+                .addEncodedPathSegment("/api/events/track")
+                .build();
+        Request request = new Request.Builder()
+                .method("POST", RequestBody.create(MediaType.parse("application/json; charset=utf-8"), "{}"))
+                .url(url)
+                .build();
+        Mockito.when(call.request()).thenReturn(request);
+        Mockito.when(call.execute()).thenThrow(java.net.SocketTimeoutException.class);
+
+        makeIterableRequest(call, UUID.randomUUID());
+        // When a timeout occurs, the logger writes a message with the encoded path
+        Mockito.verify(call.request().url().encodedPath());
```

Comment:
If an exception occurs in `makeIterableRequest` (as expected), this line will never be executed I think, so it doesn't really verify anything.

Reply:
The logger will still grab the URL and log a message since I wrapped the `.execute()` call in a block to catch the `java.net.SocketTimeoutException`. I realized that I needed to mock all of the calls to get the URL path when my original test hit a NPE 😩

Reply:
Looks like `makeIterableRequest` will throw a `RetriableError` when it encounters `java.net.SocketTimeoutException`, which is the scenario being tested here. As a result, the last line won't be executed unless the method _doesn't_ throw `RetriableError`, in which case the test case is failed already and we don't care about `call.request().url().encodedPath()`.

Reply:
Also, `call.request().url()` is not a mock, so I don't think we can verify calls on it.

Reply:
You can verify whether anything after `makeIterableRequest` gets executed or not by adding something like `assertTrue(false);` at the end. :D

Reply:
Oh I see what you mean now. I need the `Call` fixture for the log statement, but the `Mockito.verify()` line will never be executed if `makeIterableRequest` fails as expected.

Reply:
Fixed